### DISCUSSION TITLE
Create Plugin Form Interface Methods

### DISCUSSION THREAD

**Discussion by yassinebennani** [microsoft/semantic-kernel#9289] [Ideas]
Is there a way to create plugin functions from interface method definitions and then use IServiceProvider to invoke them?

Additionally, is it possible to serialize a KernelFunction to use it across services? The goal is to decentralize (distribute) plugin definitions, providing them to a service connected to an LLM, and then handle manual function invocation by calling the appropriate service to execute the function.

I hope my question is clear.

Thank you.

**Comment by sophialagerkranspandey** [comment]
Thanks for your feedback, we'd love to understand your scenario a bit more. Could you please elaborate on the use case for this? Thanks!
